core/src/main/kotlin/com/malinskiy/marathon/actor/StateMachine.kt

```diff
@@ -38,29 +38,33 @@ class StateMachine<STATE : Any, EVENT : Any, SIDE_EFFECT : Any> private construc
     private val stateRef = AtomicReference<STATE>(graph.initialState)
```

Comment:
This does not need to be an atomic reference - all accesses are protected by a synchronized section now and hence this can be just a normal field.